tiledb/sm/filter/webp_filter.h

```diff
@@ -0,0 +1,251 @@
+   */
+  Status run_forward(
+      FilterBuffer* input_metadata,
+      FilterBuffer* input,
+      FilterBuffer* output_metadata,
+      FilterBuffer* output) const;
+
+  /**
+   * Runs the filter in reverse, returning raw colorspace values to the client
+   *
+   * @param tile Current tile on which the filter is being run
+   * @param offsets_tile Offsets tile of the current tile on which the filter is
+   * being run
+   * @param input_metadata Buffer with metadata for `input`
+   * @param input Buffer with data to be filtered.
+   * @param output_metadata Buffer with metadata for filtered data
+   * @param output Buffer with filtered data (unused by in-place filters).
+   * @param config Config object for query-level parameters
+   * @return Status::Ok() on success. Throws on failure
+   */
+  Status run_reverse(
+      const Tile& tile,
+      Tile* const tile_offsets,
+      FilterBuffer* input_metadata,
+      FilterBuffer* input,
+      FilterBuffer* output_metadata,
+      FilterBuffer* output,
+      const Config& config) const override;
+
+  /**
+   * Runs the filter in reverse, returning raw colorspace values to the client
+   *
+   * @param input_metadata Buffer with metadata for `input`
+   * @param input Buffer with data to be filtered.
+   * @param output_metadata Buffer with metadata for filtered data
+   * @param output Buffer with filtered data (unused by in-place filters).
+   * @return Status::Ok() on success. Throws on failure
+   */
+  Status run_reverse(
+      FilterBuffer* input_metadata,
+      FilterBuffer* input,
+      FilterBuffer* output_metadata,
+      FilterBuffer* output) const;
+
+  /**
+   * Sets an option on the filter
+   *
+   * @param option The filter option to set
+   * @param value Value to be set for given option
+   * @return Status::Ok() on success. Throws on failure
+   */
+  Status set_option_impl(FilterOption option, const void* value) override;
+
+  /**
+   * Gets an option from the filter
+   *
+   * @param option The filter option to retrieve
+   * @param value Location to store found option value
+   * @return Status::Ok() on success. Throws on failure
+   */
+  Status get_option_impl(FilterOption option, void* value) const override;
+
+  /**
+   * Serializes filter metadata
+   *
+   * @param serializer Serializer with buffer to store metadata
+   */
+  void serialize_impl(Serializer& serializer) const override;
+
+  /**
+   * Set tile extents to be used in tile-based image compression
+   * This filter references these extents only on the forward pass during writes
+   *
+   * @param extents Extents retrieved from array Domain object
+   */
+  void set_extent(const std::vector<ByteVecValue>& extents);
+
+ private:
+  /* ********************************* */
+  /*           PRIVATE ATTRIBUTES      */
+  /* ********************************* */
+
+  float quality_;
+  WebpInputFormat format_;
+  bool lossless_;
+  std::pair<int, int> extents_;
```

Comment:
I am confused about the extent variable. It looks like you're storing an x and y tile extent here (I'm assuming that the array that you pass in should be 2D representing image data?) And the comment for set_extent doesn't go into detail why this is only 2 things. Like is the vector guaranteed to have two elements? If so that should be written down in the function docs for set_extent.

Reply:
The WebP API needs the width and height of the image along with the row stride for each row of pixels, and to support tile-based compression I used tile extents instead of dimension bounds. I'll update the comments around this a bit to clarify, but the filter would only need the Y, X extents from dimension index 0, 1 respectively so that's why there is only a need for 2 elements here. I vaguely commented on this [here](https://github.com/TileDB-Inc/TileDB/pull/3549/files/af231d3adb89f505c649013305fca3f942495623#diff-30d4a4775b76d35cfb299cf0eceec4b47be6e911a523e85f19d8ccc65cf3b51dR71-R73) but I definitely agree the use of extents should be better explained in the doxygen for `set_extents`